homeassistant/components/vodafone_station/__init__.py

```diff
@@ -34,7 +34,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
-        coordinator: VodafoneStationRouter = hass.data[DOMAIN][entry.entry_id]
+        coordinator = cast(VodafoneStationRouter, entry.runtime_data)
```

Comment:
We normally specialize the config entry to avoid these casts.

Reply:
Done